lib/src/main/java/org/openapitools/openapistylevalidator/custom/CheckOperationTagsAreDefinedRule.java

```diff
@@ -0,0 +1,45 @@
+package org.openapitools.openapistylevalidator.custom;
+
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.models.Operation;
+import org.eclipse.microprofile.openapi.models.PathItem;
+import org.eclipse.microprofile.openapi.models.tags.Tag;
+import org.openapitools.openapistylevalidator.ErrorAggregator;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This rule will check all the tags used in the operations is defined under tag section
+ */
+public class CheckOperationTagsAreDefinedRule extends OpenAPIRule {
```

Comment:
What is the difference between this rule and the previous `OperationHasTagRule` in the CLI module?

Reply:
`OperationHasTagRule` -> check each operation in the openapi spec has tags specified.
```
openapi: 3.0.1
info:
  title: ping test
  version: '1.0'
servers:
  - url: 'http://localhost:9999/'
paths:
  /ping:
    post:
      operationId: pingGet
      responses:
        '201':
          description: OK
```
`CheckOperationTagsAreDefinedRule` -> check all the tags used in the operations has been defined already.

```
openapi: 3.0.1
info:
  title: ping test
  version: '1.0'
tags:
   - name: pets
    description: Everything about your Pets
    externalDocs:
      url: http://docs.my-api.com/pet-operations.htm
servers:
  - url: 'http://localhost:9999/'
paths:
  /ping:
    post:
      operationId: pingGet
      tags: ["I'm not exist"]
      responses:
        '201':
          description: OK
```